server/publications/collections/product-publications.app-test.js

```diff
@@ -99,50 +130,58 @@ describe("Publication", function () {
         isBackorder: false
       });
 
+      merchantShop1ProductIds = [productId1, productId2, productId3];
+      merchantShop2ProductIds = [productId4];
+      primaryShopProductIds = [productId5];
+      allProductIds = [productId1, productId2, productId3, productId4, productId5];
+      activeMerchantProductIds = [productId2, productId3, productId4];
+
       collector = new PublicationCollector({ userId: Random.id() });
     });
 
     describe("Products", function () {
-      it("should return all products to admins", function (done) {
+      it("should return all products to admins in the Primary Shop", function (done) {
         // setup
-        sandbox.stub(Reaction, "getShopId", () => shopId);
+        sandbox.stub(Reaction, "getShopId", () => primaryShopId);
         sandbox.stub(Roles, "userIsInRole", () => true);
         sandbox.stub(Reaction, "hasPermission", () => true);
-        sandbox.stub(Reaction, "getShopsWithRoles", () => [shopId, primaryShopId]);
-
-        let isDone = false;
+        sandbox.stub(Reaction, "getShopsWithRoles", () => [shopId, merchantShopId, primaryShopId]);
 
         collector.collect("Products", 24, undefined, {}, (collections) => {
-          const products = collections.Products;
-
-          expect(products.length).to.equal(4);
+          const productIds = collections.Products.map(p => p._id);
 
-          if (!isDone) {
-            isDone = true;
-            done();
```

Comment:
I had been noticing that on expectation failures the errors in the terminal output cited a timeout, instead of information about the failure. 
this is because `expect` throws an error on failure, skipping the `done()` call.
`PublicationCollector#collect` returns a Promise, so we can use `Promise#then` to yield better error messages by _always_ calling `done`